common/Jet_QA.C

```diff
@@ -393,90 +418,108 @@ void CommonJetQA(std::optional<uint32_t> trg = std::nullopt)
 	// create modules that are independent of/take multiple R values ------------
 
 	// initialize and register mass, eta, and pt qa module
-	// this part is for subtracted background towers 
-	JetKinematicCheck* kinematicQA = new JetKinematicCheck(
-			"JetKinematicCheck" + trig_tag + "_towersub1_antikt",
-			JetQA::JetInput[JetQA::Type::AntiKtTowerSubR02],
-			JetQA::JetInput[JetQA::Type::AntiKtTowerSubR03],
-			JetQA::JetInput[JetQA::Type::AntiKtTowerSubR04],
-			JetQA::JetInput[JetQA::Type::AntiKtTowerSubR05]
-			);
-	kinematicQA -> Verbosity(verbosity);
-	kinematicQA -> setHistTag("");
-	kinematicQA -> setRestrictEtaRange(JetQA::RestrictEtaByR);
-	kinematicQA -> setPtRange(ptJetRange.first, ptJetRange.second);
-	kinematicQA -> setEtaRange(etaJetMaxRange.first, etaJetMaxRange.second);
-	if (trg.has_value())
+	// this part is for subtracted background towers
+	if (JetQA::HasCalos)
 	{
-		kinematicQA -> setTrgToSelect(trg.value());
-	}
-	se -> registerSubsystem(kinematicQA);
+	  if (JetQA::QA_SubtractedTowers)
+	  {	  
+	    JetKinematicCheck* kinematicQA = new JetKinematicCheck(
+	    "JetKinematicCheck" + trig_tag + "_towersub1_antikt",
+	    JetQA::JetInput[JetQA::Type::AntiKtTowerSubR02],
+	    JetQA::JetInput[JetQA::Type::AntiKtTowerSubR03],
+	    JetQA::JetInput[JetQA::Type::AntiKtTowerSubR04],
+	    JetQA::JetInput[JetQA::Type::AntiKtTowerSubR05]);
```

Comment:
Then if you go with the switch-case approach, you can `GetJetsToQA` to get the appropriate jets to QA here regardless of which mode you're in.
```suggestion
	    "JetKinematicCheck" + trig_tag + "_towersub1_antikt",
	    JetQA::JetInput[vecJetsToQA[0]],
	    JetQA::JetInput[vecJetsToQA[1]],
	    JetQA::JetInput[vecJetsToQA[2]],
	    JetQA::JetInput[vecJetsToQA[3]]);
```
(And then, another enum to index the resolutions -- e.g. `Res::R03` -- can help with readibility)